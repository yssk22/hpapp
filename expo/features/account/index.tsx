import ListItemClearCache from './internal/ListItemClearCache';

export { ListItemClearCache };
