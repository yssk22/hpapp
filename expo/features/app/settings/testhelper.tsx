import { act, render } from '@testing-library/react-native';

import { SettingsProvider, SettingsProviderProps } from './';
import SettingsList from './internals/SettingsList';

/**
 * Clear all settings
 */
export async function clearSettings() {
  await Promise.all(
    SettingsList.map((settings) => {
      settings.clear();
    })
  );
}

export async function renderSettingsProvider(children: React.ReactNode, props: SettingsProviderProps = {}) {
  await clearSettings();
  const content = render(<SettingsProvider {...props}>{children}</SettingsProvider>);
  await act(async () => {});
  return content;
}
