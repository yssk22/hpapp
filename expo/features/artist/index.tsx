import ArtistMemberFollowIcon from './internals/ArtistMemberFollowIcon';
import ArtistMemberIcon, { ArtistMemberIconSize } from './internals/ArtistMemberIcon';

export { ArtistMemberFollowIcon, ArtistMemberIcon, ArtistMemberIconSize };
