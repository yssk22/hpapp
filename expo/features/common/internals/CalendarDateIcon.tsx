import { isInvalidDate } from '@hpapp/foundation/date';
import { View, StyleSheet } from 'react-native';

import Text from './Text';

/**
 * CalendarDateIcon is an icon that displays the date of the calendar.
 */
export default function CalendarDateIcon({ date }: { date: Date | null | undefined }) {
  return (
    <View style={styles.column}>
      {!isInvalidDate(date) ? (
        <>
          <Text style={styles.textMonth}>{date!.getMonth() + 1}æœˆ</Text>
          <Text style={styles.textDate}>{date!.getDate()}</Text>
        </>
      ) : (
        <Text style={styles.textDateUndefined}>-</Text>
      )}
    </View>
  );
}

const styles = StyleSheet.create({
  column: {
    width: 40,
    height: 40,
    flexDirection: 'column',
    borderRadius: 8,
    borderWidth: 2,
    borderColor: '#cacaca',
    alignItems: 'center',
    justifyContent: 'center'
  },
  textDateUndefined: {
    fontWeight: 'bold',
    fontSize: 20,
    textAlign: 'center',
    width: '100%'
  },
  textMonth: {
    marginTop: 2,
    fontSize: 9,
    fontWeight: 'bold',
    textAlign: 'center',
    width: '100%'
  },
  textDate: {
    marginTop: -4,
    fontWeight: 'bold',
    fontSize: 20,
    textAlign: 'center',
    width: '100%'
  }
});
