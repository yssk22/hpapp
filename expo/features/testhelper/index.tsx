import {
  mockUseNavigation,
  renderUserScreen,
  renderUserComponent,
  renderGuestComponent,
  waitForUserRootRerendered
} from './internals/jest';
import { ScreenDecorator } from './internals/storybook';

export {
  mockUseNavigation,
  renderUserScreen,
  renderUserComponent,
  renderGuestComponent,
  waitForUserRootRerendered,
  ScreenDecorator
};
