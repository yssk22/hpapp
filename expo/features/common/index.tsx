import AmebloIcon from './internals/AmebloIcon';
import AssetIcon from './internals/AssetIcon';
import CalendarDateIcon from './internals/CalendarDateIcon';
import ConsentGate from './internals/ConsentGate';
import ErrorBoundary, { FallbackComponent } from './internals/ErrorBoundary';
import ExternalImage from './internals/ExternalImage';
import Initializer from './internals/Initializer';
import Link from './internals/Link';
import Loading from './internals/Loading';
import Text from './internals/Text';
import useAssetContent from './internals/useAssetContent';
import useAsync from './internals/useAsync';
import useErrorMessage from './internals/useErrorMessage';
import useIsMounted from './internals/useIsMounted';
import useMutationPromise from './internals/useMutationPromise';
import useReloadableAsync, { ReloadableAysncResult } from './internals/useReloadableAsync';
import useURICache from './internals/useURICache';

export {
  AmebloIcon,
  AssetIcon,
  CalendarDateIcon,
  ConsentGate,
  ErrorBoundary,
  ExternalImage,
  FallbackComponent,
  Initializer,
  Link,
  Loading,
  Text,
  useErrorMessage,
  useMutationPromise,
  useReloadableAsync,
  ReloadableAysncResult,
  useAssetContent,
  useAsync,
  useIsMounted,
  useURICache
};
