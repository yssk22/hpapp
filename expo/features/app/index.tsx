import { maybeDev } from '@hpapp/foundation/environment';
import { LocalMediaManagerProvider } from '@hpapp/system/media';

import AppRelayProvider from './internals/AppRelayProvider';
import { default as Root, AppRootProps as RootProps } from './internals/AppRoot';
import AppUpdateBanner from './internals/AppUpdateBanner';
import { ThemeProvider, ThemeProviderProps } from './theme';

export type AppRootProps = RootProps & ThemeProviderProps;

export function AppRoot({ screens, ...props }: AppRootProps) {
  return (
    <ThemeProvider {...props}>
      <AppRelayProvider>
        <LocalMediaManagerProvider name={maybeDev() ? 'hellofanapp-dev' : 'hellofanapp'}>
          <AppUpdateBanner />
          <Root screens={screens} />
        </LocalMediaManagerProvider>
      </AppRelayProvider>
    </ThemeProvider>
  );
}
