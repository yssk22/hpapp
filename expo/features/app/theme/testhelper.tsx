import { SettingsProviderProps } from '@hpapp/features/app/settings';
import { renderSettingsProvider } from '@hpapp/features/app/settings/testhelper';

import { ThemeProvider } from './';

export async function renderThemeProvider(children: React.ReactNode, props: SettingsProviderProps = {}) {
  return renderSettingsProvider(<ThemeProvider>{children}</ThemeProvider>, props);
}
