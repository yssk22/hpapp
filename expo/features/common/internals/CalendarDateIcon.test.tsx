import { renderUserComponent } from '@hpapp/features/testhelper';
import { composeStories } from '@storybook/react';

import * as stories from './CalendarDateIcon.stories';

describe('CalendarDateIcon.stories', () => {
  const composed = composeStories(stories);
  Object.keys(composed).forEach((key) => {
    const C = (composed as any)[key];
    test(key, async () => {
      const content = await renderUserComponent(<C />);
      expect(content).toMatchSnapshot();
    });
  });
});
