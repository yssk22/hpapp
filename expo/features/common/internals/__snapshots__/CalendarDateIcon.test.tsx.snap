// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarDateIcon.stories Basic 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onGestureCancel={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      sheetAllowedDetents="large"
      sheetCornerRadius={-1}
      sheetExpandsWhenScrolledToEdge={true}
      sheetGrabberVisible={false}
      sheetLargestUndimmedDetent="all"
      stackPresentation="push"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          collapsable={false}
          style={
            [
              {
                "flex": 1,
              },
              {
                "backgroundColor": "white",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              {
                "backgroundColor": "#ffffff",
                "flex": 1,
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <View
                style={
                  {
                    "alignItems": "center",
                    "borderColor": "#cacaca",
                    "borderRadius": 8,
                    "borderWidth": 2,
                    "flexDirection": "column",
                    "height": 40,
                    "justifyContent": "center",
                    "width": 40,
                  }
                }
              >
                <Text
                  style={
                    [
                      {
                        "color": "black",
                      },
                      {
                        "fontSize": 14,
                      },
                      {
                        "fontSize": 9,
                        "fontWeight": "bold",
                        "marginTop": 2,
                        "textAlign": "center",
                        "width": "100%",
                      },
                    ]
                  }
                >
                  10
                  æœˆ
                </Text>
                <Text
                  style={
                    [
                      {
                        "color": "black",
                      },
                      {
                        "fontSize": 14,
                      },
                      {
                        "fontSize": 20,
                        "fontWeight": "bold",
                        "marginTop": -4,
                        "textAlign": "center",
                        "width": "100%",
                      },
                    ]
                  }
                >
                  18
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backTitle=""
        backTitleVisible={false}
        backgroundColor="#0075c2"
        color="white"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="/"
        titleColor="white"
        topInsetEnabled={false}
        translucent={false}
      />
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;

exports[`CalendarDateIcon.stories InvalidDate 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onGestureCancel={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      sheetAllowedDetents="large"
      sheetCornerRadius={-1}
      sheetExpandsWhenScrolledToEdge={true}
      sheetGrabberVisible={false}
      sheetLargestUndimmedDetent="all"
      stackPresentation="push"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          collapsable={false}
          style={
            [
              {
                "flex": 1,
              },
              {
                "backgroundColor": "white",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              {
                "backgroundColor": "#ffffff",
                "flex": 1,
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <View
                style={
                  {
                    "alignItems": "center",
                    "borderColor": "#cacaca",
                    "borderRadius": 8,
                    "borderWidth": 2,
                    "flexDirection": "column",
                    "height": 40,
                    "justifyContent": "center",
                    "width": 40,
                  }
                }
              >
                <Text
                  style={
                    [
                      {
                        "color": "black",
                      },
                      {
                        "fontSize": 14,
                      },
                      {
                        "fontSize": 20,
                        "fontWeight": "bold",
                        "textAlign": "center",
                        "width": "100%",
                      },
                    ]
                  }
                >
                  -
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backTitle=""
        backTitleVisible={false}
        backgroundColor="#0075c2"
        color="white"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="/"
        titleColor="white"
        topInsetEnabled={false}
        translucent={false}
      />
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
